internal/factsengine/gathering.go

```diff
@@ -48,7 +52,15 @@ func gatherFacts(
 		g.Go(func() error {
 			var gatheringError *entities.FactGatheringError
 
-			newFacts, err := gatherer.Gather(factsRequest)
+			newFacts, err := gatherer.Gather(ctx, factsRequest)
+
+			select {
```

Comment:
question: here we are checking if the context is cancelled, right? At the end we are not waiting for it.
I think having this is equivalent:
```
newFacts, err := gatherer.Gather(groupCtx, factsRequest)
ctxErr := ctx.Err()
switch {
case ctxErr != nil:
    return ctxErr;
case err == nil:
    factsCh <- newFacts
...

// Or simply above the `switch`
if err := ctx.Err(); err != nil {
    return err
}
```
`ctx.Err` will return not nil if it is cancelled. Just to reuse the `switch`

Reply:
Considering the select is blocking this could be another valid approach

Reply:
Done in https://github.com/trento-project/agent/pull/360/commits/ab5a951135e076bc44c808f9b1d907c0e2a7d420